google/cloud/spanner/testing/random_database_name.cc

```diff
@@ -13,17 +13,31 @@
 // limitations under the License.
 
 #include "google/cloud/spanner/testing/random_database_name.h"
+#include "google/cloud/internal/format_time_point.h"
+#include <ctime>
 
 namespace google {
 namespace cloud {
 namespace spanner_testing {
 inline namespace SPANNER_CLIENT_NS {
+
+std::string RandomDatabasePrefixRegex() {
+  return R"re("^db-\d{4}-\d{2}-\d{2}-)re";
+}
+
+std::string RandomDatabasePrefix(std::chrono::system_clock::time_point tp) {
```

Comment:
FWIW, if we wanted to use absl/time/time.h, this would look like:
```cc
std::string RandomDatabasePrefix(std::chrono::system_clock::time_point tp) {
  return absl::FormatTime("db-%Y-%m-%d-", absl::FromChrono(tp), absl::UTCTimeZone());
}
```
IMO, as long as we're not talking about public interface types, I think Abseil Time adds a lot of value and simplicity (but I realize I'm biased).